examples/solana_native/basic_nft/README.md

```diff
@@ -0,0 +1,423 @@
+      - name: gem
+        type: GemMetadata
+        solana:
+          attributes: [ init ]
+          seeds:
+            mint: mint
+      - name: color
+        type: string
+      - name: rarity
+        type: string
+      - name: short_description
+        type: string
+  - name: transfer
+    uses:
+      - csl_spl_assoc_token.create
+      - csl_spl_token.transfer_checked
+    inputs:
+      - name: mint
+        type: csl_spl_token.Mint
+      - name: gem
+        type: GemMetadata
+        solana:
+          attributes: [ mut ]
+          seeds:
+            mint: mint
+  - name: burn
+    uses:
+      - csl_spl_token.burn
+    inputs:
+      - name: mint
+        type: csl_spl_token.Mint
+      - name: gem
+        type: GemMetadata
+        solana:
+          attributes: [ mut ]
+          seeds:
+            mint: mint
+```
+
+### 2. Generate the Solana program and client library
+
+Open the terminal and type the following command
+
+```shell
+codigo solana generate nft.yaml
+```
+
+After generating the Solana program and client library, two new directories will be created relative to the `nft.yaml` file named `program_client` and `program`.
+
+### 3. Implement the business logic
+
+In the `program` directory, you will find a directory called `src`; inside this directory, there will be two `.rs` files named `mint.rs`, `burn.rs` and `transfer.rs`
+
+#### Mint business logic
+
+Copy and paste the following code just below the comment line `// Implement your business logic here...` in the file `mint.rs`
+
+```rust
+gem.data.color = color;
+gem.data.rarity = rarity;
+gem.data.short_description = short_description;
+gem.data.mint = *mint.info.key;
+gem.data.assoc_account = Some(*assoc_token_account.key);
+
+csl_spl_token::src::cpi::initialize_mint_2(for_initialize_mint_2, 0, *wallet.key, None)?;
+csl_spl_assoc_token::src::cpi::create(for_create)?;
+csl_spl_token::src::cpi::mint_to(for_mint_to, 1)?;
+csl_spl_token::src::cpi::set_authority(for_set_authority, 0, None)?;
+```
+
+#### Burn business logic
+
+Copy and paste the following code just below the comment line `// Implement your business logic here...` in the file `burn.rs`
+
+```rust
+gem.data.assoc_account = None;
+csl_spl_token::src::cpi::burn(for_burn, 1)?;
+```
+
+#### Transfer business logic
+
+Copy and paste the following code just below the comment line `// Implement your business logic here...` in the file `transfer.rs`
+
+```rust
+gem.data.assoc_account = Some(*destination.key);
+
+// Create the ATA account for new owner if it hasn't been created
+if assoc_token_account.lamports() == 0 {
+    csl_spl_assoc_token::src::cpi::create(for_create)?;
+}
+
+csl_spl_token::src::cpi::transfer_checked(for_transfer_checked, 1, 0)?;
+```
+
+### 4. Build and deploy the program
+
+Open the terminal and navigate to the `program` directory, from there execute the following command:
+
+```shell
+cargo build-sbf
+```
+If it gives an error then run the following command:
```

Comment:
Based on the above comment, we can remove this text (including the suggested command)